diff --git a/node_modules/koa-mongo-crud/src/crud/mapper.js b/node_modules/koa-mongo-crud/src/crud/mapper.js
index 6702aa2..2b73556 100644
--- a/node_modules/koa-mongo-crud/src/crud/mapper.js
+++ b/node_modules/koa-mongo-crud/src/crud/mapper.js
@@ -199,11 +199,8 @@ class CrudMapper {
       update = { $set: data };
     }
 
-    const result = await this.collection.findOneAndUpdate(filter, update, { returnOriginal: false, upsert: false });
-    if (result.ok !== 1) {
-      return null;
-    }
-    return result.value;
+    const result = await this.collection.findOneAndUpdate(filter, update, { returnDocument: "after", upsert: false });
+    return result;
   }
 
   async delete(id, userId = null) {
@@ -218,14 +215,10 @@ class CrudMapper {
     const result = await this.collection.findOneAndUpdate(
       { _id: id, deleted: { $ne: true } },
       { $set: data },
-      { returnOriginal: false, upsert: false },
+      { returnDocument: "after", upsert: false },
     );
 
-    if (result.ok !== 1) {
-      return null;
-    }
-
-    return result.value;
+    return result;
   }
 
   async remove(id) {
@@ -233,7 +226,7 @@ class CrudMapper {
 
     const result = await this.collection.findOneAndDelete(filter);
 
-    return result.ok === 1;
+    return result != null;
   }
 
   toJson(data) {
